web3.js/src/layout.ts

```diff
@@ -152,6 +152,9 @@ export function getAlloc(type: any, fields: any): number {
       if (Array.isArray(field)) {
         return field.length * getItemAlloc(item.elementLayout);
       }
+    } else if ('fields' in item) {
```

Comment:
This will match both `Structure` and `BitStructure` objects, but I _did not test it_ with `BitStructure`. I presume that `BitStructure` objects can't be of variable length, so they will match the `if (item.span >= 0)` check above.